src/ssmDocument/ssm/ssmClient.ts

```diff
@@ -0,0 +1,125 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*!
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import * as path from 'path'
+import * as nls from 'vscode-nls'
+
+const localize = nls.loadMessageBundle()
+
+import { ExtensionContext, LanguageConfiguration, languages, window, workspace } from 'vscode'
+
+import {
+    DidChangeConfigurationNotification,
+    LanguageClient,
+    LanguageClientOptions,
+    NotificationType,
+    ServerOptions,
+    TransportKind,
+} from 'vscode-languageclient'
+
+namespace ResultLimitReachedNotification {
+    export const type: NotificationType<string, any> = new NotificationType('ssm/resultLimitReached')
+}
+
+interface Settings {
+    ssm?: {
+        format?: { enable: boolean }
+        resultLimit?: number
+    }
+}
+
+export async function activate(extensionContext: ExtensionContext) {
+    const toDispose = extensionContext.subscriptions
+
+    // The server is implemented in node
+    const serverModule = extensionContext.asAbsolutePath(path.join('dist/src/ssmDocument/ssm/', 'ssmServer.js'))
+    // The debug options for the server
+    // --inspect=6010: runs the server in Node's Inspector mode so VS Code can attach to the server for debugging
```

Comment:
What happens if port 6010 is taken?

Reply:
If port 6010 is taken, the debugger won't be able to launch.